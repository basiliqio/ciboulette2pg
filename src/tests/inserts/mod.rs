use super::*;

mod inserts_success;
