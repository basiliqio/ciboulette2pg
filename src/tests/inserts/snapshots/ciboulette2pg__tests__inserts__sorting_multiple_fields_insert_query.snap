---
source: src/tests/inserts/mod.rs
expression: "sqlformat::format(query.as_str(), &sqlformat::QueryParams::None,\n                  sqlformat::FormatOptions::default())"

---
WITH "cte_peoples_insert_0" AS (
  INSERT INTO
    "public"."peoples" (
      "age",
      "first-name",
      "gender",
      "last-name",
      "twitter"
    )
  VALUES
    ($1, $2, $3, $4, $5) RETURNING *
),
"cte_peoples_data_1" AS (
  SELECT
    "cte_peoples_insert_0"."id" :: TEXT AS "id",
    "cte_peoples_insert_0"."id" AS "main_id",
    NULL :: TEXT AS "related_id",
    NULL :: TEXT AS "related_type",
    $6 :: TEXT AS "type",
    JSON_BUILD_OBJECT(
      $7,
      "cte_peoples_insert_0"."age",
      $8,
      "cte_peoples_insert_0"."first-name",
      $9,
      "cte_peoples_insert_0"."gender",
      $10,
      "cte_peoples_insert_0"."last-name",
      $11,
      "cte_peoples_insert_0"."twitter"
    ) AS "data",
    "cte_peoples_insert_0"."favorite_color" AS "rel_favorite_color",
    "cte_peoples_insert_0"."first-name" AS "sort_first-name",
    "cte_peoples_insert_0"."age" AS "sort_age"
  FROM
    "cte_peoples_insert_0"
),
"cte_final_main_data" AS (
  SELECT
    DISTINCT ON ("cte_peoples_data_1"."main_id") "cte_peoples_data_1".*,
    "cte_peoples_data_1"."sort_first-name" AS "sort__first-name",
    "cte_peoples_data_1"."sort_age" AS "sort__age"
  FROM
    "cte_peoples_data_1"
  ORDER BY
    "cte_peoples_data_1"."main_id"
) (
  SELECT
    "cte_final_main_data"."id",
    "cte_final_main_data"."type",
    "cte_final_main_data"."data",
    "cte_final_main_data"."related_type",
    "cte_final_main_data"."related_id"
  FROM
    "cte_final_main_data"
  ORDER BY
    "cte_final_main_data"."sort__age" ASC,
    "cte_final_main_data"."sort__first-name" DESC
);
